### PR TITLE
TEXT-98: Remove isDelimiter and use HashSets for delimiter checks

### PR DESCRIPTION
**Pull request by arunvinudss**
@chtompki Please review and let me know what you think?

### CODE REVIEW

File: src/main/java/org/apache/commons/text/WordUtils.java [hunk 1]
```diff
@@ -747,45 +750,29 @@ public static boolean containsAllWords(final CharSequence word, final CharSequen
...
+     * value if delimiters is null. The generated hash set provides O(1) lookup time.
+     * </p>
      *
-     * @param ch  the character to check
-     * @param delimiters  the delimiters
-     * @return true if it is a delimiter
+     * @param delimiters set of characters to determine capitalization, null means whitespace
+     * @return Set<Integer>
      */
-    public static boolean isDelimiter(final char ch, final char[] delimiters) {
```
Comment (ameyjadiye) [inline src/main/java/org/apache/commons/text/WordUtils.java]: Rather removing we should keep this method.

File: src/main/java/org/apache/commons/text/WordUtils.java [hunk 2]
```diff
@@ -747,45 +750,29 @@ public static boolean containsAllWords(final CharSequence word, final CharSequen
...
 
-  //-----------------------------------------------------------------------
-    /**
-     * Is the codePoint a delimiter.
-     *
-     * @param codePoint the codePint to check
-     * @param delimiters  the delimiters
-     * @return true if it is a delimiter
-     */
-    public static boolean isDelimiter(final int codePoint, final char[] delimiters) {
```
Comment (ameyjadiye) [inline src/main/java/org/apache/commons/text/WordUtils.java]: Rather removing we should keep this method.

**Comment by arunvinudss** [comment]
@ameyjadiye I want to remove the isDelimiter method. I would be surprised if someone uses isDelimiter separately because all it does is to check if a given element is present in an array or not. Moreover the isDelimiter char version is already dead code as we don't use it anymore. I would say the scope of the isDelimiter should have been private.

**Comment by PascalSchumacher** [comment]
@arunvinudss While I agree that `isDelimiter` should have been private, it is public and was released with commons-text `1.1`. Due to the strict binary compatibilities promise of commons it can not be removed before `2.0`. For now the best we can do is mark it as deprecated and explain that it will be removed in version `2.0`.

**Comment by ameyjadiye** [comment]
@arunvinudss , just addition to @PascalSchumacher comment , at this point we don't know other than Commons text who else having dependancy on `isDelimiter` so better we can make it depricated and we can remove code all together in 2.x

**Comment by chtompki** [comment]
This all opens the question about going `2.x`. I think we have a couple of things that would warrant a 2.x move. Do we want to attempt that in the fall, or is it still too premature?

**Comment by ameyjadiye** [comment]
@chtompki , I think whichever items are piled up for 2.x are not too critical, we should wait for 2.x release.
If we are releasing some major improvement or fix we can release all queued items in that. ATM I don't see anything.

**Comment by arunvinudss** [comment]
@chtompki Please review and merge.

**Comment by chtompki** [comment]
Will get to this today.

**Pull request merged**